src/lycophron/tasks/tasks.py

```diff
@@ -16,24 +16,49 @@
 # TODO tasks are using 'db' directly. Breaks the basic flow Interface -> Business -> Data.
 # TODO tasks do not have any logging implemented yet (e.g. success, errors).
 # TODO tasks are not updating the record's local status (e.g. 'SUCCESS', 'FAILED', etc)
+# TODO record serialization to zenodo is done in place, should have its own module (e.g. marshmallow serializers)
+# TODO status codes are not verified. E.g. only upload records whose deposit are uploaded, only upload deposits that have not been processed before etc
 
+class SqlAlchemyTask(Task):
+    """An abstract Celery Task that ensures that the connection the the
+    database is closed on task completion"""
 
-@app.task
-def process_all_files():
-    pass
+    abstract = True
 
+    def after_return(self, status, retval, task_id, args, kwargs, einfo):
+        db.session.remove()
 
-@app.task
-def create_deposit(files, metadata, token, url):
+
+@app.task(base=SqlAlchemyTask)
+def create_deposit(record, token, url):
     """Create a new deposit."""
+    doi = record.get("doi")
+    record_id = record["id"]
+    files = record.get("files", [])
+    metadata = record.get("original", {})
+    communities = record.get("communities")
+
     # Create deposit
     session = create_session(token)
     r = session.post(url, json={})
-    # db.update_record_status(doi, "DEPOSIT_CREATED")
+
+    # TODO uncomment when MemoryError is fixed
+    # db.update_record_status(record_id, "DEPOSIT_CREATED")
```

Comment:
Interacting with the DB inside a celery task raises a `MemoryError`. This is important because there's no "status" for each record right now. We're blindly uploading the records. 

This is bad because e.g. some records might already been uploaded and we do not want to upload them again.